### PR TITLE
Add a sample for running PowerCLI as "Cloud Shell".

### PR DESCRIPTION
**Pull request by mattmoor**

### CODE REVIEW

File: samples/cloud-power-shell/README.md [hunk 1]
```diff
@@ -0,0 +1,119 @@
+## Using `VSphereBinding` to create a `PowerCLI` "Cloud Shell"
+
+This sample builds on our [previous sample](../powercli/README.md) to show how
+to use `VSphereBinding` to create a `PowerCLI` "Cloud Shell" by running the
+`vmware/powerclicore` container image as a Knative Service.
+
+
```
Comment (mattmoor) [inline samples/cloud-power-shell/README.md:7]: Format markdown:
```suggestion

```

File: samples/cloud-power-shell/README.md [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+### Pre-requisites
+
+This sample assumes that you have a vSphere environment set up already with
+credentials in a Secret named `vsphere-credentials`. For the remainder of the
+sample we will assume you are within the environment setup for the
+[`vcsim` sample](../vcsim/README.md).
+
+
```
Comment (mattmoor) [inline samples/cloud-power-shell/README.md:15]: Format markdown:
```suggestion

```

File: samples/cloud-power-shell/README.md [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+### Create the Binding
+
+We are going to use the following binding to authenticate our `PowerCLI`
+"Cloud Shell":
```
Comment (mattmoor) [inline samples/cloud-power-shell/README.md:19]: Format markdown:
```suggestion
We are going to use the following binding to authenticate our `PowerCLI` "Cloud
Shell":
```

File: samples/cloud-power-shell/README.md [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+```shell
+kubectl apply -f binding.yaml
+```
+
+### Building our "Cloud Shell" service.
+
+For the "shell" part of our demo, we are going to make use of
+[yudai/gotty](https://github.com/yudai/gotty).  We are going to use the
+following `ko` configuration (in `.ko.yaml`) to base `gotty` on
+`vmware/powerclicore`:
```
Comment (mattmoor) [inline samples/cloud-power-shell/README.md:55]: Format markdown:
```suggestion
[yudai/gotty](https://github.com/yudai/gotty). We are going to use the following
`ko` configuration (in `.ko.yaml`) to base `gotty` on `vmware/powerclicore`:
```

File: samples/cloud-power-shell/README.md [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+        args:
+        # Tell gotty to enable interacting with the session.
+        - -w
+        # Launch Powershell and run our setup commands without exiting.
+        - pwsh
+        - -NoExit
+        - -Command
+        - |
+          Set-PowerCLIConfiguration -InvalidCertificateAction Ignore -Confirm:$false | Out-Null
+          Connect-VIServer -Server ([System.Uri]$env:GOVC_URL).Host -User $env:GOVC_USERNAME -Password $env:GOVC_PASSWORD
```
Comment (mattmoor) [inline samples/cloud-power-shell/README.md:89]: Format markdown:
```suggestion
        # The binary is gotty (based on vmware/powerclicore)
        - image: ko://github.com/vmware-tanzu/sources-for-knative/vendor/github.com/yudai/gotty
          args:
            # Tell gotty to enable interacting with the session.
            - -w
            # Launch Powershell and run our setup commands without exiting.
            - pwsh
            - -NoExit
            - -Command
            - |
              Set-PowerCLIConfiguration -InvalidCertificateAction Ignore -Confirm:$false | Out-Null
              Connect-VIServer -Server ([System.Uri]$env:GOVC_URL).Host -User $env:GOVC_USERNAME -Password $env:GOVC_PASSWORD
```

File: samples/cloud-power-shell/README.md [hunk 6]
```diff
@@ -0,0 +1,119 @@
...
+```
+
+This Service authenticates `PowerCLI` using our injected credentials, and then
+creates a session over a websocket that allows the user to interact with `PowerCLI`
+over a websocket.  You can deploy this service via:
```
Comment (mattmoor) [inline samples/cloud-power-shell/README.md:94]: Format markdown:
```suggestion
creates a session over a websocket that allows the user to interact with
`PowerCLI` over a websocket. You can deploy this service via:
```

**Review by mattmoor — COMMENTED** [review]
Produced via:
  `prettier --write --prose-wrap=always $(find -name '*.md' | grep -v vendor | grep -v .github | grep -v docs/cmd/)`

File: samples/cloud-power-shell/README.md [hunk 7]
```diff
@@ -0,0 +1,119 @@
...
+
+```shell
+kubectl get ksvc cloud-power-shell
+```
+
+When it reports as `Ready`, open the URL in your browser and try running:
+
+```shell
+Get-VIEevent | Write-Host
+````
```
Comment (mattmoor) [inline samples/cloud-power-shell/README.md:110]: Format markdown:
```suggestion
```
```

File: samples/cloud-power-shell/README.md (hunk 1 above)
Comment (mattmoor) [inline samples/cloud-power-shell/README.md:7]: Format markdown:
```suggestion

```

File: samples/cloud-power-shell/README.md (hunk 2 above)
Comment (mattmoor) [inline samples/cloud-power-shell/README.md:15]: Format markdown:
```suggestion

```

File: samples/cloud-power-shell/README.md (hunk 3 above)
Comment (mattmoor) [inline samples/cloud-power-shell/README.md:19]: Format markdown:
```suggestion
We are going to use the following binding to authenticate our `PowerCLI` "Cloud
Shell":
```

File: samples/cloud-power-shell/README.md (hunk 4 above)
Comment (mattmoor) [inline samples/cloud-power-shell/README.md:55]: Format markdown:
```suggestion
[yudai/gotty](https://github.com/yudai/gotty). We are going to use the following
`ko` configuration (in `.ko.yaml`) to base `gotty` on `vmware/powerclicore`:
```

File: samples/cloud-power-shell/README.md (hunk 5 above)
Comment (mattmoor) [inline samples/cloud-power-shell/README.md:89]: Format markdown:
```suggestion
        # The binary is gotty (based on vmware/powerclicore)
        - image: ko://github.com/vmware-tanzu/sources-for-knative/vendor/github.com/yudai/gotty
          args:
            # Tell gotty to enable interacting with the session.
            - -w
            # Launch Powershell and run our setup commands without exiting.
            - pwsh
            - -NoExit
            - -Command
            - |
              Set-PowerCLIConfiguration -InvalidCertificateAction Ignore -Confirm:$false | Out-Null
              Connect-VIServer -Server ([System.Uri]$env:GOVC_URL).Host -User $env:GOVC_USERNAME -Password $env:GOVC_PASSWORD
```

**Review by mattmoor — COMMENTED** [review]
Produced via:
  `prettier --write --prose-wrap=always $(find -name '*.md' | grep -v vendor | grep -v .github | grep -v docs/cmd/)`

File: samples/cloud-power-shell/README.md (hunk 6 above)
Comment (mattmoor) [inline samples/cloud-power-shell/README.md:94]: Format markdown:
```suggestion
creates a session over a websocket that allows the user to interact with
`PowerCLI` over a websocket. You can deploy this service via:
```

**Review by vaikas — APPROVED** [review]

**Pull request merged**